lib/puppet/type/group.rb

```diff
@@ -203,7 +203,7 @@ def delimiter
     # @return [Boolean] if the group exists on the system
     # @api private
     def exists?
-      provider.exists?
+      provider && provider.exists?
```

Comment:
Was the provider being `nil` something you ran into in the rspec tests, or something during a "real" run?

Reply:
The service_user_available? and service_user_available? methods end up calling this function and on the server side, provider isn't set so it would die with a NilClass does not have method '.exists?'